File: data/code_pairs/cpp/138.cpp
#include <vector>
#include <string>

struct asdf_record {
    std::vector<float> record;
};

struct asdf_event {
    std::string event = std::string(13, ' '); // Ensure the string is of length 13, filled with spaces
    int nreceivers;
    std::vector<std::string> receiver_name_array;
    std::vector<std::string> network_array;
    std::vector<std::string> component_array;
    std::vector<std::string> receiver_id_array;
    std::vector<asdf_record> records;
};

File: data/code_pairs/fortran/138.f90
module asdf_data

  type asdf_record
    real, pointer :: record(:)
  end type asdf_record

  type asdf_event
    character(len=13)     :: event
    integer           :: nreceivers
    character(len=20), pointer :: receiver_name_array(:), network_array(:)
    character(len=20), pointer :: component_array(:)
    character(len=20), pointer :: receiver_id_array(:)
    type (asdf_record), pointer :: records(:)
  end type asdf_event

end module asdf_data